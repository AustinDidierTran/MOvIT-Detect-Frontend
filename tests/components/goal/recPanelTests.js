/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import React from 'react';
import PropTypes from 'prop-types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import RecPanel from '../../../src/components/goal/recPanel';

Enzyme.configure({ adapter: new Adapter() });

describe('RecPanel Tests', () => {
  const props = {
    condition: true,
    title: 'This is a test',
    value: 10,
  };

  it('should have proptypes', () => {
    const actualValue = RecPanel.propTypes;

    const expectedValue = {
      condition: PropTypes.bool.isRequired,
      title: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<RecPanel {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
