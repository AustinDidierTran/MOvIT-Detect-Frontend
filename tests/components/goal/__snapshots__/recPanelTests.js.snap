// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecPanel Tests should match the snapshot 1`] = `
<div
  className="col-12 col-sm-6 col-md-4"
  style={
    Object {
      "card": Object {
        "height": "100%",
      },
      "padding": "0px",
    }
  }
>
  <CustomCard
    element={
      <span>
        10
      </span>
    }
    header={
      <div
        className="ui-card-title"
      >
        This is a test
      </div>
    }
    style={
      Object {
        "height": "100%",
      }
    }
  />
  <Tooltip
    appendTo="body"
    escape={true}
    for="#recPanelundefined"
    hideDelay={null}
    onBeforeShow={null}
    positionStyle={null}
    showDelay={null}
    title={null}
    tooltipDisabled={false}
    tooltipEvent="hover"
    tooltipPosition="right"
    tooltipStyleClass={null}
  />
</div>
`;
